pkg/kubelet/deviceplugin/device_plugin_stub.go

```diff
@@ -70,7 +70,7 @@ func (m *Stub) Start() error {
 	// Wait till grpc server is ready.
 	for i := 0; i < 10; i++ {
```

Comment:
This whole loop is useless. `m.server.GetServiceInfo()` returns the number of services as soon as RegisterDevicePluginServer is called

Reply:
I think it returns the number of services when they are ready, not as soon as RegisterDevicePluginServer is called.

Reply:
https://github.com/grpc/grpc-go/blob/master/server.go#L376

You can also test this

Reply:
Hmm, this is interesting to know. I thought GetServiceInfo returns ready-to-serve no. of methods, but looking at the code, it is indeed should return No. of methods right after registration. However, I did verify that if I removed this code block, running 'bazel test --runs_per_test=20 //pkg/kubelet/deviceplugin:go_default_test' became flaky (I think mostly on endpoint_test.go) but with the current code, the tests all passed. I think we will need to spend more time understand this code. For now, I would like to keep it this way to make sure the tests are not flaky.